### PR TITLE
Use `docker-compose` for production and development

### PR DESCRIPTION
**Pull request by enrico-ghidoni**
Added docker-compose files for production and development environments and moved each env configuration into `.dev` files to keep secrets, well, secret as well as improve flexibility.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# ---				NOTE						---
+# This is a stub file, all credentials and connection strings should be replaced
+# with actual values. Do not fill in this file, instead make a `.env.XYZ` file 
+# and fill the values in the new one to keep environments separated. Lastly, pass 
+# the correct env file to docker-compose with the --env-file flag.
+
+# - Infrastructure parameters
+
+ROOSTER_IMAGE_TAG=0.5.0
```
Comment (milivojm) [inline .env:14]: As we talked about this, maybe just a `preview` tag is ok.

File: .env.dev [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+RABBITMQ_HOST=roster-rabbitmq
+RABBITMQ_USERNAME=guest
+RABBITMQ_PASSWORD=guest
+
+# -- Database configuration
+CONNECTION_STRINGS_ROSTER=Host=postgres-dev;Port=5432;Database=roster;Username=postgres;Password=cnto_dev
+CONNECTION_STRINGS_POSTGRES=Host=postgres-dev;Port=5432;Database=identity;Username=postgres;Password=cnto_dev
+
+# -- MailJet configuration
+MAILJET_BASE_URL=https://rooster.carpenoctem.co
```
Comment (milivojm) [inline .env.dev]: `http://localhost:5000`

File: .env.dev [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+MAILJET_KEY=436a142b035f598a3d882116e4ece3af
```
Comment (milivojm) [inline .env.dev]: Think we should not expose our API keys on GitHub. :)

File: .env.dev (hunk 2 above)
Comment (milivojm) [reply .env.dev]: My bad: http://localhost:8902

**Review by milivojm — COMMENTED** [review]
```
[mmilani@fedora rooster]$ sudo docker ps
[sudo] password for mmilani: 
CONTAINER ID   IMAGE                                COMMAND                  CREATED          STATUS          PORTS                                                                                                           NAMES
a74a0a46b068   cntoarma/rooster:0.5.0               "dotnet Roster.Web.d…"   17 minutes ago   Up 17 minutes   0.0.0.0:8902->80/tcp, :::8902->80/tcp                                                                           cnto-roster
e80bb371605d   cntoarma/rooster-db:0.5.0-snapshot   "docker-entrypoint.s…"   18 minutes ago   Up 17 minutes   0.0.0.0:5432->5432/tcp, :::5432->5432/tcp                                                                       postgres-dev
fddee4869393   masstransit/rabbitmq:3.9.7           "docker-entrypoint.s…"   18 minutes ago   Up 17 minutes   4369/tcp, 5671-5672/tcp, 15671/tcp, 15691-15692/tcp, 25672/tcp, 0.0.0.0:15672->15672/tcp, :::15672->15672/tcp   roster-rabbitmq
```

Consistent versioning might be good, whatever way you pick it's fine. 

Good work!

File: .env (hunk 1 above)
Comment (enrico-ghidoni) [reply .env:14]: Yep and I thought we agreed on keeping versions 0.X.X until full deployment :D

File: .env.dev (hunk 2 above)
Comment (enrico-ghidoni) [reply .env.dev]: Good spotting! Thanks

Comment (enrico-ghidoni) [reply .env.dev:23]: Can you build `0.5.0` for this? I don't think I have all the migrations yet

File: .env.dev (hunk 3 above)
Comment (enrico-ghidoni) [reply .env.dev]: My bad, thought `docker-compose.yml` with the tokens was already on GitHub

**Review by milivojm — APPROVED** [review]

**Pull request merged**